src/ecdsa/ecdsa.py

```diff
@@ -171,7 +171,13 @@ def sign(self, hash, random_k):
     G = self.public_key.generator
     n = G.order()
     k = random_k % n
-    p1 = k * G
+    # Fix the bit-length of the random nonce,
+    # so that it doesn't leak via timing.
+    # This does not change that ks = k mod n
+    ks = k + n
+    if ks.bit_length() == n.bit_length():
+      ks = ks + n
```

Comment:
why it's added twice? it won't change its value `mod n`...

Reply:
That's exactly the point, `ks = k mod n`, yet the bit-length of `ks` is constant. See https://eprint.iacr.org/2011/232 and their mitigation in the Conclusion section.